frappe/core/doctype/doctype/doctype.py

```diff
@@ -246,6 +247,20 @@ def validate_website(self):
 			# clear website cache
 			clear_cache()
 
+	def validate_max_attachment_count(self):
+		"""Ensure that max_attachments is *at least* bigger than number of attach fields"""
+		from frappe.model import attachment_fieldtypes
+
+		if not self.max_attachments:
+			return
+
+		total_attach_fields = len([d for d in self.fields if d.fieldtype in attachment_fieldtypes])
+		if total_attach_fields > self.max_attachments:
+			self.max_attachments = total_attach_fields
+
+		frappe.msgprint(_("Number of attachment fields exceeds limit, updated max attachments limit to {}.")
```

Comment:
Shouldn't this throw instead? I mean, if the goal is to ensure the action 🤔

Reply:
Why throw? 🤔 We are already changing the value on L259. The message is just to let user/dev know that it was silently updated

Reply:
My bad, I read the method name and docstring and assumed this was supposed to be a validation.

---

Some pointers after looking through this properly;
* The `frappe.msgprint` should be run only when the value is updated...So, inside the `if total_attach_fields > self.max_attachments:` block
* Update the docstring and method name to something more appropriate
* The message should be updated this because reason, not the other way around. Also, the labels should be used from the meta instead? for consistency [opinion]
* This message and validation should happen on the Desk instead? Triggering a msgprint during Document save might also trigger user confusion. (UX here feels mid-2000s) [opinion]

Reply:
Moved whole thing to client side JS. 100s of kittens died to bring us this change!